ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/SubscriptionManager.java

```diff
@@ -158,4 +161,20 @@ public void sendMessage(final Long subscriptionId, final JsonRpcResult msg) {
           }
         });
   }
+
+  @Override
+  public void onPrivateTransactionProcessed(final PrivateTransactionEvent event) {
+    // When a user is removed from a privacy group, remove all subscriptions from that user to that
+    // group
+    subscriptionsOfType(SubscriptionType.LOGS, PrivateLogsSubscription.class).stream()
```

Comment:
Do you need to handle the same race condition as filters (eg user removed from group but then in the same block requests to create a new subscription) - which we solved by storing the private tx events until the block was processed.

Reply:
PR updated to address this.